deepfence_server/ingesters/agent.go

```diff
@@ -237,168 +238,139 @@ func concatMaps(input map[string][]string) []map[string]interface{} {
+			if !ok {
+				continue
+			}
+			if len(adjacency) == 0 {
+				inbound_edges = append(inbound_edges,
+					map[string]string{"destination": hostName, "left_pid": "0", "right_pid": strconv.Itoa(n.Pid)})
+			} else {
+				edges := make([]map[string]string, 0, len(adjacency))
+				for _, i := range adjacency {
+					if n.NodeID != i {
+						ip, port := extractIPPortFromEndpointID(i)
+						if host, ok := resolvers.get_host(ip); ok {
+							if hostName == host {
+								continue
+							}
+							right_ippid, ok := resolvers.get_ip_pid(ip + port)
+							if ok {
+								rightpid := extractPidFromNodeID(right_ippid)
+								edges = append(edges,
+									map[string]string{"destination": host, "left_pid": strconv.Itoa(n.Pid), "right_pid": rightpid})
+							}
+						} else {
 							edges = append(edges,
-								map[string]string{"destination": host, "left_pid": node_info["pid"], "right_pid": rightpid})
+								map[string]string{"destination": "out-the-internet", "left_pid": strconv.Itoa(n.Pid), "right_pid": "0"})
 						}
-					} else {
-						edges = append(edges,
-							map[string]string{"destination": "out-the-internet", "left_pid": node_info["pid"], "right_pid": "0"})
 					}
-					//endpoint_edges = append(endpoint_edges, map[string]string{"left": node_info["node_id"], "right": i})
 				}
+				endpoint_edges_batch = append(endpoint_edges_batch, map[string]interface{}{"source": hostName, "edges": edges})
 			}
-			endpoint_edges_batch = append(endpoint_edges_batch, map[string]interface{}{"source": host_name, "edges": edges})
 		}
 	}
-
-	// Handle inbound from internet
-	edges := []map[string]string{}
-	for _, n := range rpt.Endpoint.Nodes {
-		node_info := n.ToDataMap()
-		if _, ok := node_info["host_node_id"]; !ok {
-			node_ip, _ := extractIPPortFromEndpointID(node_info["node_id"])
-			if val, ok := resolvers.get_host(node_ip); ok {
-				node_info["host_node_id"] = val
-			} else {
-				// This includes skipping all endpoint having 127.0.0.1
-				continue
-			}
-		}
-		//node_info["adja"] = strings.Join(n.Adjacency, ",")
-		//endpoint_batch = append(endpoint_batch, node_info)
-
-		host_name := extractHostFromHostNodeID(node_info["host_node_id"])
-		if len(node_info["pid"]) > 0 && len(n.Adjacency) == 0 {
-			edges = append(edges,
-				map[string]string{"destination": host_name, "left_pid": "0", "right_pid": node_info["pid"]})
-		}
+	if len(inbound_edges) > 0 {
+		endpoint_edges_batch = append(endpoint_edges_batch, map[string]interface{}{"source": "in-the-internet", "edges": inbound_edges})
 	}
-	endpoint_edges_batch = append(endpoint_edges_batch, map[string]interface{}{"source": "in-the-internet", "edges": edges})
 
-	process_batch := make([]map[string]string, 0, len(rpt.Process.Nodes))
+	process_batch := make([]map[string]interface{}, 0, len(rpt.Process))
 	process_edges_batch := map[string][]string{}
-	for _, n := range rpt.Process.Nodes {
-		node_info := n.ToDataMap()
-		host := node_info["node_id"]
```

Comment:
What is the value of `NodeID` here?
When constructing the graph, we expect the `node_id` of process to be in a specific format: `host_name;pid` to reconstruct connections quickly

Reply:
Process node_id format unchanged https://github.com/deepfence/ThreatMapper/blob/a53772ad0618b72dc8cbd25f7516808abd1bad15/deepfence_agent/tools/apache/scope/probe/process/reporter.go#L123